src/titiler/extensions/titiler/extensions/wms.py

```diff
@@ -0,0 +1,534 @@
+                                "bounds_wgs84"
+                            ] = src_dst.geographic_bounds
+                            layers_dict[layer]["abstract"] = src_dst.info().json()
+
+                # Build information for the whole service
+                minx, miny, maxx, maxy = zip(
+                    *[layers_dict[layer]["bounds_wgs84"] for layer in layers_dict]
+                )
+
+                return templates.TemplateResponse(
+                    f"wms_{version}.xml",
+                    {
+                        "request": request,
+                        "request_url": wms_url,
+                        "formats": self.supported_format,
+                        "available_epsgs": self.supported_crs,
+                        "layers_dict": layers_dict,
+                        "service_dict": {
+                            "xmin": min(minx),
+                            "ymin": min(miny),
+                            "xmax": max(maxx),
+                            "ymax": max(maxy),
+                        },
+                    },
+                    media_type=MediaType.xml.value,
+                )
+
+            # GetMap: Return an image chip
+            if request_type.lower() == "getmap":
+                # Required parameters:
+                # - VERSION
+                # - REQUEST=GetMap,
+                # - LAYERS
+                # - STYLES
+                # - CRS
+                # - BBOX
+                # - WIDTH
+                # - HEIGHT
+                # - FORMAT
+                # Optional parameters: TRANSPARENT, BGCOLOR, EXCEPTIONS, TIME, ELEVATION, ...
+
+                # List of required parameters (styles and crs are excluded)
+                req_keys = {
+                    "version",
+                    "request",
+                    "layers",
+                    "bbox",
+                    "width",
+                    "height",
+                    "format",
+                }
+
+                intrs = set(req.keys()).intersection(req_keys)
+                missing_keys = req_keys.difference(intrs)
+                if len(missing_keys) > 0:
+                    raise HTTPException(
+                        status_code=400,
+                        detail=f"Missing 'GetMap' parameters: {missing_keys}",
+                    )
+
+                version = req["version"]
+                if version not in self.supported_version:
+                    raise HTTPException(
+                        status_code=400,
+                        detail=f"Invalid 'VERSION' parameter: {version}. Allowed versions include: {self.supported_version}",
+                    )
+
+                if not set(req.keys()).intersection({"crs", "srs"}):
+                    raise HTTPException(
+                        status_code=400, detail="Missing 'CRS' or 'SRS parameters."
+                    )
+
+                crs = CRS.from_user_input(req.get("crs", req.get("srs")))
```

Comment:
Should we raise HTTPException if crs ends up being empty?

Reply:
we could 👍